tests/handlers/test_user_directory.py

```diff
@@ -119,19 +124,22 @@ def test_handle_user_deactivated_support_user(self):
                 user_id=s_user_id, password_hash=None, user_type=UserTypes.SUPPORT
             )
         )
-
-        self.store.remove_from_user_dir = Mock(return_value=defer.succeed(None))
-        self.get_success(self.handler.handle_local_user_deactivated(s_user_id))
-        self.store.remove_from_user_dir.not_called()
+        with patch.object(
+            self.store, "remove_from_user_dir", return_value=defer.succeed(None)
+        ) as mock:
+            self.get_success(self.handler.handle_local_user_deactivated(s_user_id))
+        self.get_success(mock.not_called())
 
     def test_handle_user_deactivated_regular_user(self):
         r_user_id = "@regular:test"
         self.get_success(
             self.store.register_user(user_id=r_user_id, password_hash=None)
         )
-        self.store.remove_from_user_dir = Mock(return_value=defer.succeed(None))
-        self.get_success(self.handler.handle_local_user_deactivated(r_user_id))
-        self.store.remove_from_user_dir.called_once_with(r_user_id)
+        with patch.object(
+            self.store, "remove_from_user_dir", return_value=defer.succeed(None)
+        ) as mock:
+            self.get_success(self.handler.handle_local_user_deactivated(r_user_id))
+        self.get_success(mock.called_once_with(r_user_id))
```

Comment:
... or a `called_once_with` method?

Reply:
I guess the original author was missing an `assert_` from the method names.

Reply:
Hm, this is concerning (and it was very possibly me, though I can't remember). Is there any way we can do this that will properly result in a failure if a typo is made?

Reply:
Different typing stubs for `Mock` maybe---removing its `__getattr__`?

Reply:
as of python 3.7, you can call `mock.seal` (https://docs.python.org/3/library/unittest.mock.html#sealing-mocks). But that doesn't really help you now, and I think it just replaces one thing to forget with another thing to forget.